EIPS/eip-5320.md

```diff
@@ -0,0 +1,156 @@
+
+1. Alice owns an Item, and values at an amount X. 
+2. Alice pays taxes according to a tax rate R to a collector, depending on this self-assesed value.
+3. Bob offers Alice an amount Y. Because Y is higher than X, Alice is forced to sell the Item to him.
+4. Bob self-asseses the value too at Z, and now pays taxes over it.
+
+This mechanic sets a stage in which actors won't overvalue the item to avoid paying excessive taxes, but won't undervalue the item either, to avoid a third party to force them to sell it.
+
+The main obvious case of Harberger Taxes is to be used as a coordination mechanism for land allocation, but, to give some examples:
+
+- Real-world land allocation.
+- Digital land allocation (videogames, social experiments...).
+- Advertisement Space in websites.
+- Governance mechanisms that use "seats" subject to Harberger.
+- Social media that requires holding a token to interact.
+
+## Motivation
+
+Provide smart contract interoperability to interact with Harberger Taxes, which is a pricing mechanism that can provide utility to many different applications.
+
+## Specification
+
+Every contract implementing this EIP must implement the `ERC5320` and `ERC721` interfaces.
+
+```solidity
+interface ERC5320 is ERC721 {
+
+  /// @dev emits when a token is forcibly bought through the Harberger mechanism.
+  /// @param _tokenId The identifier of the token that was bought.
+  /// @param _owner The owner of the token after the purchase.
+  /// if equal to address(0), it means the token was revoked.
+  /// @param _amount The amount that was paid in this transaction.
+  event TokenBought(uint256 indexed _tokenId, address indexed _owner, uint256 _amount);
+
+  /// @dev emits when an token is funded with currency to pay the taxes, or when this
+  /// fund was updated.
+  /// @param _tokenId The identifier of the token that had a fund change.
+  /// @param _amount The updated amount of the fund.
+  event TokenFunded(uint256 indexed _tokenId, uint256 _amount);
+
+  /// @dev emits when the valuation of an token changes.
+  /// @param _tokenId The identifier of the token whose valuation changed.
+  /// @param _valuation The new valuation of the token.
+  event ValuationSet(uint256 indexed _tokenId, uint256 _valuation);
+  
+  /// @dev emits when a token has tax paid for them.
+  /// @param _tokenId The identifier of the token that was taxed.
+  /// @param _value The taxes that have been paid.
+  event TaxPaid(uint256 indexed _tokenId, uint256 _value);
+
+  /// @dev buy a token. This function must succeed if 
+  /// @param _tokenId The identifier of the token to buy.
+  /// @param _offer The maximum amount offered to buy this token.
+  /// @param _valuation The valuation the token will have after a successful purchase.
+  /// @param _fund The fund put in to pay for the taxes of the token.
+  function buy(uint256 _tokenId, uint256 _offer, uint256 _valuation, uint256 _fund) external;
```

Comment:
The specification should make it explicit that when this function is called, `collect` must be called internally before the sale completes.